src/domain/usecases/GetDataSetSettingsUseCase.ts

```diff
@@ -26,29 +26,31 @@ export class GetDataSetSettingsUseCase {
             return Future.success({
                 coreCompetencies,
                 dataSet,
-                indicators: this.config.indicators.map(indicator => {
-                    const existingIndicator = dataSet.indicators.find(
-                        ind => ind.id === indicator.id
-                    );
-                    return existingIndicator
-                        ? Indicator.create({
-                              ...indicator,
-                              disaggregation:
-                                  existingIndicator.type === "outputs"
-                                      ? existingIndicator.disaggregation
-                                      : undefined,
-                              relatedDataElements:
-                                  existingIndicator.type === "outcomes"
-                                      ? existingIndicator.relatedDataElements
-                                      : [],
-                          })
-                        : indicator;
-                }),
-                existingIndicatorsIds: dataSet.indicators.map(indicator => indicator.id),
+                indicators: this.getIndicatorsFromDataSet(dataSet),
+                existingIndicatorIds: dataSet.indicators.map(indicator => indicator.id),
```

Comment:
nothing to change, but in some projects we create `getId` (in Ref.ts) to make this very common usage more compact.